### DISCUSSION TITLE
How to open a hostless pcm stream (IV feedback) in SOF without intervention of middle ware?

### DISCUSSION THREAD

**Discussion by Venkata-Prasad-Potturu** [thesofproject/sof#9476] [Q&A]
Hi All,
We are working on speaker protection solution.
How to get open IV feedback stream when we start playback on Speaker endpoint without intervention of Pipewire?
How does SOF Framework supports hostless pcm streams?

**Comment by ranj063** [comment]
@Venkata-Prasad-Potturu do you need to a open a separate capture stream? Or do you just want to enable the feedback path?

**Comment by Venkata-Prasad-Potturu** [comment]
We just want to enable feedback path when playback starts on smart amp speaker.
That feedback IV data goes to smart amp component running in the DSP for speaker protection.

**Reply by lgirdwood** [reply]
@Venkata-Prasad-Potturu have you had a look at existing Intel topologies for IPC4? There are some smart amp examples, but I cant remember if they use a user space helper stream for IV. Failing that, it may be worth creating a hostless consumer/producer endpoint module that uses a kcontrol to activate/deactivate the IV stream (this is in our todo list, but we haven't got around to it yet).
